flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorTestBase.java

```diff
@@ -215,7 +211,7 @@ Source<Integer, MockSourceSplit, Set<MockSourceSplit>> createMockSource() {
 
     protected SourceCoordinatorContext<MockSourceSplit> getNewSourceCoordinatorContext() {
         return new SourceCoordinatorContext<>(
-                coordinatorExecutor,
+                Executors.newScheduledThreadPool(1, coordinatorThreadFactory),
                 Executors.newScheduledThreadPool(
                         1,
                         new ExecutorThreadFactory(
```

Comment:
Can you check that each created coordinator is actually properly closed? I think there are places:
org.apache.flink.runtime.source.coordinator.SourceCoordinatorTest#testSerdeBackwardCompatibility
org.apache.flink.runtime.source.coordinator.SourceCoordinatorConcurrentAttemptsTest#testCoordinatorThrowExceptionIfWatermarkAlignmentIsEnabled

Where this is not true, and your change would cause a thread leak? Maybe instead of fixing each place individually, we can drop `sourceCoordinator.close()` from `@AfterEach` method above, and just create a `CloseableRegistry`, where this method `getNewSourceCoordinatorContext` would register every newly created registry and we could clean reigstry in `@AfterEach`?

Reply:
Thanks for the review!

Good catch! updated.